test/stock-count.spec.js

```diff
@@ -1,95 +1,60 @@
-    // Check that the products are available in stock count xform
-    const productCategoryList = stockCountScenario.productCategoryScenario;
-    const productsList = stockCountScenario.productsScenario;
-    const workbook = new ExcelJS.Workbook();
-    await workbook.xlsx.readFile(formPath);
-    const surveyWorkSheet = workbook.getWorksheet('survey');
-    const nameCol = surveyWorkSheet.getColumn('B');
-    const cellProductCategoriesList = [];
-    const cellProductsList = [];
-    let productCatIndex =0;
-    let productIndex = 0;
-    nameCol.eachCell(function(cell){
-
-      if(cell.value === productCategoryList[productCatIndex] && productCatIndex < productCategoryList.length){
-        cellProductCategoriesList.push(cell.value);
-        productCatIndex ++;
-        productIndex = 0;
-      }
-
-      if(cell.value === productsList[productIndex] && productIndex < productsList.length){
-        cellProductsList.push(cell.value);
-        productIndex ++;
-      }
-      
-    });
-
-    expect(productsList.length).toBe(cellProductsList.length);
-    expect(productsList.entries).toStrictEqual(cellProductsList.entries);
-    
-    expect(productCategoryList.length).toBe(cellProductCategoriesList.length);
-    expect(productCategoryList.entries).toStrictEqual(cellProductCategoriesList.entries);
-
-    //Removing the stock monitoring init file and stock count file
-    expect(fs.unlinkSync(stockMonitoringConfig)).toBe(undefined);
-    expect(fs.unlinkSync(formPath)).toBe(undefined);
-    expect(fs.unlinkSync(formPropertiesPath)).toBe(undefined);
-
-    const translationFiles = fs.readdirSync(path.join(processDir, 'translations'));
-    for(const translationFile of translationFiles){
-      
-      const messageFileContent = fs.readFileSync(path.join(processDir, 'translations', translationFile), {encoding: 'utf-8'});
-      expect(messageFileContent).not.toBe('');
-      const newMessageContent = messageFileContent.split('\n').map(message => {
-        if(!message.toString().includes('cht-stock-monitoring-workflow') && message.toString()!==''){
-          return `${message.toString()}\n`;
-        }
-      });
-
-      expect(newMessageContent.includes('cht-stock-monitoring-workflow')).toBe(false);
-      fs.truncate(path.join(processDir, 'translations', translationFile), 0, function () {});
-      fs.writeFile(path.join(processDir, 'translations', translationFile),newMessageContent.toString().replaceAll(',', ''));
-      
+
+    for(const createdAppFormFile of createdAppFormFiles){
+      expect(fs.existsSync(path.join(processDir, 'forms', 'app', createdAppFormFile))).toBe(true);
     }
-  });
 
-});
+    // Check that the products and categories are available in stock count xform
+    const { productsList, productCategoryList } = await readDataFromXforms(stockCountScenario.productCategoryScenario, stockCountScenario.productsScenario, 'stock_count.xlsx');
 
+    expect(productsList.length).toBe(stockCountScenario.productsScenario.length);
```

Comment:
Should we verify the name of the products/categories?

Reply:
I am comparing the entries here `expect(productsList.entries).toStrictEqual(stockCountScenario.productsScenario.entries);`.

This is should check the name and any other values in the both arrays.